### PR TITLE
FAPI: Add http return code handling

### PR DESCRIPTION
**Pull request by AndreasFuchsTPM**
Add handling of the http return code to the downloading of the EK intermediate CA certificates.

Signed-off-by: Andreas Fuchs <andreas.fuchs@infineon.com>

### CODE REVIEW

File: src/tss2-fapi/ifapi_curl.c [hunk 1]
```diff
@@ -435,6 +436,12 @@ ifapi_get_curl_buffer(unsigned char * url, unsigned char ** buffer,
         goto out_easy_cleanup;
     }
 
+    curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+    if (http_code >= 400) {
```
Comment (williamcroberts) [inline src/tss2-fapi/ifapi_curl.c:440]: `curl_easy_perform` returns `CURLE_HTTP_RETURNED_ERROR` if the error code is `>= 400`, I don't see how this check adds anything?

See:
  - https://curl.se/libcurl/c/curl_easy_perform.html
  - https://curl.se/libcurl/c/libcurl-errors.html

**Review by williamcroberts — CHANGES_REQUESTED** [review]

File: src/tss2-fapi/ifapi_curl.c (hunk 1 above)
Comment (AndreasFuchsTPM) [reply src/tss2-fapi/ifapi_curl.c:440]: Hmmm... That's weird.
I have a system here where the CA is not online and I get this:
```
ERROR:fapi:src/tss2-fapi/ifapi_curl.c:193:ifapi_curl_verify_ek_cert() Failed to convert intermediate certificate to X509 format. (size=1245): (then the segfault)
```

AFAIK this should not happen if curl_easy_perform() returns an error.

With the patch, it does work.

**Review by JuergenReppSIT — APPROVED** [review]
I could also observe this behaviour with libcurl4-openssl-dev:amd64. An error is only returned if CURLOPT_FAILONERROR is set. So the PR should be merged.

File: src/tss2-fapi/ifapi_curl.c (hunk 1 above)
Comment (williamcroberts) [reply src/tss2-fapi/ifapi_curl.c:440]: Sounds like a bug in curl, but sometimes we just gotta code around those :-p

**Review by williamcroberts — APPROVED** [review]

**Comment by williamcroberts** [comment]
> I could also observe this behaviour with libcurl4-openssl-dev:amd64. An error is only returned if CURLOPT_FAILONERROR is set. So the PR should be merged.

Should we have set that option instead?

**Comment by JuergenReppSIT** [comment]
> Should we have set that option instead?

imho the version from the pr is ok and it's not necessary.

**Pull request merged**